### ISSUE TITLE
[a11y]: IconButton repeating label when used in Table component in Firefox browser

### ISSUE THREAD

**Issue by michael-c-ibm** [carbon-design-system/carbon#21073]
### Package

@carbon/react

### Browser

Firefox

### Operating System

MacOS

### Package version

^1.68.0

### React version

^16.12.0

### Automated testing tool and ruleset

Manual Testing

### Assistive technology

VoiceOver

### Description

```
<TableCell className="incident-reports-download-button">
   <IconButton
   align="left"
   kind="primary"
   className="cds--btn cds--btn--ghost cds--btn--sm"
   label={
   // 'cds--visually-hidden' carbon class is used to only display this text for screen readers
   <>
   <span className="cds--visually-hidden">
   {row.title}
   </span>
   {t('incident-reports.download-button')}
   </>
   }
   onClick={() => {
   window.location.assign(
   `/status/getCustomerIncidentReport?fileName=${row.fileName}`,
   );
   }}
   >
   <Download aria-hidden="true" />
   </IconButton>
</TableCell>
```

I have a table cell that contains an `IconButton` component. When I press tab, it cycles across every `Download` icon. To improve accessibility, I want it to name the `title`of the row I'm in + the word `download`.

For example:
`Power Outage report - Download, Button`

To do this, I set the label to contain a react fragment with a hidden title + my translated download text. This way the actual title text is not shown visually, and only seen by VoiceOver.

The problem I'm facing, is that this works on Chrome. However in Firefox, there is a bug where for the first half of the rows, it will read the label, but then also include the previous row's download text, then repeat the current download text again.

I've tried manually overriding the `aria-labbeledby` property but was unsuccessful. I've also tried using the `Button` component as well with `hasIconOnly` prop. But because label sets the `aria-labbeledby` property under the hood, both attempts were unsuccessful.

What's interesting is that after the first half, it will read out properly.

https://github.com/user-attachments/assets/fdcf747a-dd2f-48e2-af16-44b1f030e109
Works in Firefox^

https://github.com/user-attachments/assets/6c0e80a1-c253-4396-b27a-59edc2aaa3fc
Does not work in Chrome^

### Reproduction/example

https://stackblitz.com/edit/github-76wvhdk8?file=src%2FApp.jsx

### Steps to reproduce

1. Create table with 7 rows with IconButton components at the end of every table row
2. For each IconButton, set the label to be a react fragment like this:
```
<IconButton
                label={
                  <>
                    <span className="cds--visually-hidden">ignore text</span>
                    add
                  </>
                }
              >
                <Add />
              </IconButton>
```

Where it contains a span that is hidden visually and its value is the title of the row
And it contains the text `add`

3. Turn on VoiceOver and tab across the table
4. Note the first half of the table will repeat the previous row's add button label text, then repeat the current row's add button label text

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
